### DISCUSSION TITLE
CSP settings are being violated in some Mantine components

### DISCUSSION THREAD

**Discussion by igomest** [mantinedev/mantine#5777] [Q&A]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.5.1

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

I used the `react-helmet-async` library to configure CSP in my project. Additionally, I am dynamically generating a nonce for each page update.

I'm using Mantine's getStyleNonce function to apply the nonce to the style tags. However, when changing to dark theme or opening a modal, drawer, etc, the nonce is violated. Even though it was applied using getStyleNonce and React Helmet.

Can you help me with the solution?

```jsx
import '@mantine/core/styles.css';
import { MantineProvider, Modal, Button } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { Router } from './Router';
import { theme } from './theme';
import Buffer from 'buffer';

import { Helmet } from 'react-helmet-async';
import { Drawer } from './Drawer';

const generateNonce = () => {
  const buffer = Buffer.Buffer.from(crypto.randomUUID());
  return buffer.toString('base64');
};

const nonce = generateNonce();

export default function App() {
  const [opened, { open, close }] = useDisclosure(false);

  return (
    <MantineProvider theme={theme} getStyleNonce={() => nonce}>
      <Helmet>
        <meta
          httpEquiv="Content-Security-Policy"
          content={`object-src 'none'; upgrade-insecure-requests; default-src 'self'; connect-src ws: http://localhost:5173/; style-src 'self' 'nonce-${nonce}'; block-all-mixed-content; form-action 'self'; script-src 'self' 'nonce-${nonce}' http://localhost:5173/`}
        ></meta>
      </Helmet>
      <Modal opened={opened} onClose={close} title="Authentication">
        {/* Modal content */}
      </Modal>

      <Button onClick={open}>Open modal</Button>

      <Router />

      <Drawer />
    </MantineProvider>
  );
}

```

### If possible, include a link to a codesandbox with a minimal reproduction

https://github.com/igomest/csp-mantine-error

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Based on your function, the generated nonce is random and executes on every render. At the same time, I suppose that helmet does not update the meta in the head when other components rerender. To fix this issue, ensure that your csp strategy includes these dynamic values or use static values.
